python_ta/debug/accumulation_table.py

```diff
@@ -0,0 +1,158 @@
+"""
+Table data structure that prints a nicely formatted table
+for an accumulator loop
+"""
+from __future__ import annotations
+
+import inspect
+import sys
+
+import astroid
+import tabulate
+
+
+def no_of_whitespaces(with_line: str) -> int:
+    """Return the number of whitespaces that are in the line of code
+    containing the with statement
+    """
+    blank_chars = 0
+    for char in with_line:
+        if char == " ":
+            blank_chars += 1
+        else:
+            break
+
+    return blank_chars
+
+
+def loop_string(full_string: list[str], no_whitespaces: int) -> str:
```

Comment:
Similar to above comment, rename this function and reword the description to capture the full generality of the behaviour. This function returns the lines up to the first line...

Rename `full_string` (to something like `lines`) and `no_whitespaces` (see above comment about "no").